pkg/kubelet/envvars/envvars.go

```diff
@@ -22,28 +22,30 @@ import (
 	"strconv"
 	"strings"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
 )
 
 // FromServices builds environment variables that a container is started with,
 // which tell the container where to find the services it may need, which are
-// provided as an argument.
-func FromServices(services []*v1.Service) []v1.EnvVar {
+// provided as an argument. "masterFilterFunc" checks if the service is the master
+// service and it can be of type ExternalName.
+func FromServices(services []*v1.Service, masterFilterFunc func(*v1.Service) bool) []v1.EnvVar {
 	var result []v1.EnvVar
 	for i := range services {
 		service := services[i]
 
 		// ignore services where ClusterIP is "None" or empty
 		// the services passed to this method should be pre-filtered
-		// only services that have the cluster IP set should be included here
-		if !v1helper.IsServiceIPSet(service) {
+		// only services that have the cluster IP set should be included here.
+		// Exception is made for the master service, which can be of type "ExternalName".
+		if !v1helper.IsServiceIPSet(service) && !masterFilterFunc(service) {
```

Comment:
Instead of limiting this to the Kubernetes master service only with masterFilterFunc, why not remove this check entirely and allow all `ExternalName` services to populate these environment variables?

See https://github.com/kubernetes/kubernetes/issues/60535

Reply:
Mainly due to https://github.com/kubernetes/kubernetes/issues/60535#issuecomment-490757630

Reply:
My suggestion to ease those concerns, https://github.com/kubernetes/kubernetes/issues/60535#issuecomment-594186643

Reply:
Feel free to resolve this conversation. After some consideration I need this change only for `KUBERNETES_SERVICE_HOST` and `KUBERNETES_SERVICE_PORT`